aws-fargate/fargate.yaml

```diff
@@ -0,0 +1,287 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: CloudFormation template for a SpiceDB cluster.
+Parameters:
+  VPC:
+    Type: AWS::EC2::VPC::Id
+  SubnetA:
+    Type: AWS::EC2::Subnet::Id
+  SubnetB:
+    Type: AWS::EC2::Subnet::Id
+  Certificate:
+    Type: String
+    # Update with the certificate ARN from Certificate Manager, which must exist in the same region.
+    # In our case, it is permissions-staging.domain.com
+    Default: 'arn:aws:acm:us-east-1:5xxxxxxxxx3:certificate/6e603b3c-....-....-....-72b1d8d4711b'
+  Image:
+    Type: String
+    # Update with the Docker image. "You can use images in the Docker Hub registry or specify other repositories (repository-url/image:tag)."
+    Default: 5xxxxxxxxx3.dkr.ecr.us-east-1.amazonaws.com/permissions-staging:latest
+  ServiceName:
+    Type: String
+    # update with the name of the service
+    Default: perms-stg-service
+  ContainerPort:
+    Type: Number
+    Default: 50051
+  LoadBalancerPort:
+    Type: Number
+    Default: 50051
+  HealthCheckPath:
+    Type: String
+    Default: '/grpc.health.v1.Health/Check'
+  HostedZoneName:
+    Type: String
+    Default: domain.com
+  Subdomain:
+    Type: String
+    Default: permissions-staging
+  # for autoscaling
+  MinContainers:
+    Type: Number
+    Default: 1
+  # for autoscaling
+  MaxContainers:
+    Type: Number
+    Default: 2
+  # target CPU utilization (%)
+  AutoScalingTargetValue:
+    Type: Number
+    Default: 50
```

Comment:
autoscaling could be disruptive to a spicedb cluster, because the ring has to reconfigure and it would potentially affect your deployment's cache hit ratios

Reply:
My initial suggestion was not to enable autoscaling. While SpiceDB supports adding and removing nodes, it's something that comes at a cost of reorganizing the hash ring, and leads to lower cache hit rate. However, after discussing with the team, there are things we could explore to reduce the impact of instances of SpiceDB coming and going, and we certainly want to make autoscaling fully supported without any performance impact.

So you can dismiss my comment!